src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/serializer.html.twig

```diff
@@ -0,0 +1,184 @@
+{% extends '@WebProfiler/Profiler/layout.html.twig' %}
+
+{% block toolbar %}
+    {% set icon %}
+        {{ include('@WebProfiler/Icon/serializer.svg') }}
+    {% endset %}
+
+    {% set text %}
+        <div class="sf-toolbar-info-piece">
+            Serializer
+        </div>
+    {% endset %}
+
+    {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', { 'link': true }) }}
+{% endblock %}
+
+{% block menu %}
+    {% set disable = collector.serializations | length == 0 and collector.deserializations | length == 0 %}
+    <span class="label {{ disable ? 'disabled' }}">
+        <span class="icon">{{ include('@WebProfiler/Icon/serializer.svg') }}</span>
+        <strong>Serializer</strong>
+    </span>
+{% endblock %}
+
+{% block panel %}
+    <h2>Serializer</h2>
+
+    <div class="metrics">
+        <div class="metric">
+            <span class="value">{{ collector.serializations | length }}</span>
+            <span class="label">Serializations</span>
+        </div>
+        <div class="metric">
+            <span class="value">{{ collector.deserializations | length }}</span>
+            <span class="label">Deserializations</span>
+        </div>
+        <div class="metric">
+            <span class="value">{{ collector.normalizations | length }}</span>
+            <span class="label">Normalizations</span>
+        </div>
+        <div class="metric">
+            <span class="value">{{ collector.denormalizations | length }}</span>
+            <span class="label">Denormalizations</span>
+        </div>
+    </div>
+
+    <div class="sf-tabs">
+        <div class="tab {{ collector.serializations|length == 0 ? 'disabled' }}">
+            <h3 class="tab-title">Serializations <span class="badge">{{ collector.serializations | length }}</span></h3>
+
+            <div class="tab-content">
+                <h4>Serializations</h4>
+                <table>
+                    <thead>
+                    <tr>
+                        <th>#</th>
+                        <th>Format</th>
+                        <th>From</th>
+                        <th>To</th>
+                        <th>Context</th>
+                    </tr>
+                    </thead>
+                    <tbody>
+                    {% for serialization in collector.serializations %}
+                        <tr>
+                            <td class="font-normal text-small text-muted nowrap">{{ loop.index }}</td>
+                            <td class="nowrap">{{ serialization.format }}</td>
+                            <td>{{ profiler_dump(serialization.data, maxDepth=2) }}</td>
+                            <td>{{ serialization.result }}</td>
```

Comment:
What is the type of `$result`? shouldn't you use `profiler_dump`?

Reply:
The result of a serialization should always be a string, shouldn't it? Otherwise profiler_dump makes sense here, too.